src/EditorFeatures/CSharpTest/Completion/CompletionServiceTests.cs

```diff
@@ -25,5 +32,55 @@ public void AcquireCompletionService()
             var service = CompletionService.GetService(document);
             Assert.NotNull(service);
         }
+
+        [Theory, CombinatorialData]
+        public async Task GettingCompletionListShoudNotRunSourceGenerator(bool forkBeforeFreeze)
+        {
+            var sourceMarkup = @"
+using System;
+
+namespace N
+{
+    public class C1
+    {
+        $$
+    }
+}";
+            MarkupTestFile.GetPosition(sourceMarkup.NormalizeLineEndings(), out var source, out int? position);
+
+            var generatorRanCount = 0;
+            var generator = new CallbackGenerator(onInit: _ => { }, onExecute: _ => { generatorRanCount++; });
```

Comment:
Do an Interlocked.Increment() here to avoid any potential race. It's not _hugely_ important since you're asserting that the count is simply zero at the end, so even if we were to somehow run in parallel the test would fail; I worry that either it'll get copy/pasted, or we'll have a bug where it's running in parallel and confuse whoever is trying to debug the test.